antiope-s3/src/Antiope/S3/Types.hs

```diff
@@ -33,6 +36,14 @@ data S3Uri = S3Uri
   , objectKey :: ObjectKey
   } deriving (Show, Eq, Generic)
 
+instance FromText S3Uri where
+  parser = do
+    _  <- DAT.string "s3://"
+    bn <- BucketName . T.pack <$> DAC.many1 (DAT.satisfy (\c -> c /= '/' && c /= ' '))
```

Comment:
Perhaps change the `/= ' '` to some sort of generic doesn't-match-whitespace?

Reply:
Turns out space is valid.  Does that mean s3 uris aren't really uris?

Reply:
I think it can follow the hadoop FS specs... But I am not sure

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html

Reply:
Space is listed as " might require additional code handling and likely will need to be URL encoded or referenced as HEX"

Reply:
I think it's amazonka's responsibility to encode it in that case.

Reply:
If Amazonka expects "raw" text as the parameter to `ObjectKey` then sure.